doc/index.md

```diff
@@ -236,114 +292,131 @@ local myconnection = http_client.connect {
+
 
 ### `h1_connection:read_request_line(timeout)` <!-- --> {#http.h1_connection:read_request_line}
 
+Gets the request method, requested path and HTTP version for an incoming request. On error, returns nil, an error message and an error number.
+
 
 ### `h1_connection:read_status_line(timeout)` <!-- --> {#http.h1_connection:read_status_line}
 
+Gets the HTTP version (1 or 1.1) , status code and a reason description if applicable. If the socket could not be read, returns nil, an error message and an error number.
+
 
 ### `h1_connection:read_header(timeout)` <!-- --> {#http.h1_connection:read_header}
 
+Returns the HTTP header (?key, val) from an incoming request. If the header could not be read, the function returns nil an error and an error message.
+
 
 ### `h1_connection:read_headers_done(timeout)` <!-- --> {#http.h1_connection:read_headers_done}
 
+Checks if all chunks of the HTTP headers have been fully received. Returns nil, an error and an error message if the socket cannot be read.
+
 
 ### `h1_connection:read_body_by_length(len, timeout)` <!-- --> {#http.h1_connection:read_body_by_length}
 
+Get the number of bytes from the buffered HTTP request body. Use a negative number for *up to* that number of bytes. Asserts if len is not a number.
+
 
 ### `h1_connection:read_body_till_close(timeout)` <!-- --> {#http.h1_connection:read_body_till_close}
 
+Reads the entire request body. This function will yield until complete or `timeout` is expired.
+
 
 ### `h1_connection:read_body_chunk(timeout)` <!-- --> {#http.h1_connection:read_body_chunk}
 
+Reads the next available line of data from the request and returns the chunk and it's size. This function will yield until complete or `timeout`. Returns nil, and 0 if no data was read. Returns nil, an error and an error message if there was an error reading.
+
 
 ### `h1_connection:write_request_line(method, path, httpversion, timeout)` <!-- --> {#http.h1_connection:write_request_line}
 
+Sends an HTTP request to a remote server. Yields until success or `timeout`. If the write fails, returns nil, an error message and an error number.
+
 
 ### `h1_connection:write_status_line(httpversion, status_code, reason_phrase, timeout)` <!-- --> {#http.h1_connection:write_status_line}
 
+Sends a status response to the remote connection. Yields until timeout. If the write fails, the funtion returns nil, an error message and an error number. 
+
 
 ### `h1_connection:write_header(k, v, timeout)` <!-- --> {#http.h1_connection:write_header}
 
+Writes a key:value string to the socket. Yields until success or `timeout`. Returns nil, an error message and an error if the write fails. 
+
 
 ### `h1_connection:write_headers_done(timeout)` <!-- --> {#http.h1_connection:write_headers_done}
 
+Finalizes the http headers in the socket. Yields until success or `timeout`. Returns nil, an error message and an error if the write fails.
```

Comment:
'Finalizes' has special meaning in programming.

Better to say: Terminates a header block by sending `"\r\n"`